### DISCUSSION TITLE
Automatically select by customer color inquiry

### DISCUSSION THREAD

**Discussion by seleondevel** [kimai/kimai#4924] [Plugins]
Hello.

Thank you for your help so far.

I am using Kimai well as Sprint schedule management tool, thank you.

However, there are some regrets, so I'm giving you an opinion.

If you check the part shown in the capture image below, it would be nice to change it as follows.

![스크린샷 2024-06-21 104649](https://github.com/kimai/kimai/assets/165639563/cee93a9b-84cd-4826-b7ec-6c31162f974b)
![스크린샷 2024-06-21 104732](https://github.com/kimai/kimai/assets/165639563/1dd5b912-edc8-4c5f-b579-436f006c5324)

1. When you create a project from the customer menu, the color is automatically selected as the customer color automatically

2. If you select customer in the Project creation window, color automatically changes to customer color

3. When you create an activity from the project menu, color is automatically selected as the customer color set for the project

By any chance, can't you reflect it in the next kimai version update?

@kevinpapst I'd like to hear your ideas on this.

Please reply.

**Comment by kevinpapst** [comment]
I won't change the current behavior. 
But there is an existing issue where we discuss how to improve the color handling. 

If you want it to adapt it for your use-case, you can use form events (or *PreCreateEvents) to change the color.

**Comment by seleondevel** [comment]
public function onProjectCreateEvent(ProjectCreateEvent $event): void
 {
        $this->logger->info(sprintf('onProjectCreateEvent!!'));

        $builder = $this->createFormBuilder(ProjectType::class);
        ...............
}

@kevinpapst  
*PreCreateEvent does not work when you run through the Project Creation or Activity Creation window.

 As shown above in ProjectCreateEvent or ActivityCreateEvent, 

$builder = $this->createFormBuilder (ProjectType:::class); 
or $builder = $this->createFormBuilder (ActivityType::class); 

I try to automatically set up color in the creation window, but I don't know how to access the form data.

Or, should I implement the code in a different way?

Can I ask you some advice on how to implement it in the code above?
